### ISSUE TITLE
Changes to reports to prepare for the report generator.

### ISSUE THREAD

**Issue by teryhill** [LibreHealthIO/LibreEHR#998]
The reports us basically the same functions, Date to, Date from, Provider select, facility select etc. 

Task is to pull those functions out of the reports and create a library file named report_functions.php. Place the functions in report_functions.php.

@pri2si17-1997

**Comment by Trodrige** [comment]
@pri2si17-1997 and @teryhill I think this is good for me to handle.

**Comment by aethelwulffe** [comment]
Report selection criteria and display function library...common use, and maybe user-extensible.

**Comment by sandipbhuyan** [comment]
Which functions are needed to moved from the report section php files.

**Comment by Trodrige** [comment]
> Report selection criteria and display function library...common use, and maybe user-extensible.

I don't get you here @aethelwulffe.

**Comment by teryhill** [comment]
the provider and facility are inputs that may be using functions. The date to and from are the start and end dates and are done in the reports with no calls to any functions.

**Comment by NirmalKumarJoshi** [comment]
Hello @teryhill can I work on resolving this issue?

**Comment by aethelwulffe** [comment]
@NirmalKumarJoshi How about networking with @Trodrige in this issue thread, and maybe start a communal branch adding functions that can be called by reporting tools?  Trodrige has push permissions on this repo, and can start a feature branch that you can fork and make commits to.
You good with that T-rod?

**Comment by Trodrige** [comment]
@teryhill do you want me to pull out the functions that these variables ($date_to, $date_from, $facility, $provider) use and put in reports_functions.php? I'm confused here. Some of the functions used by these variables are found in other library files like patient.inc and formatting.inc

**Comment by teryhill** [comment]
This is from the collections report.  Turn this into a function and then call it from the reports that use it. For the same with the dates

![image](https://user-images.githubusercontent.com/4655366/36846968-b870f9c4-1d2a-11e8-98f4-2143ac9815f9.png)

**Comment by aethelwulffe** [comment]
Yeah, there are some functions floating around in various includes.  Those header files are messy themselves.  All function include files should have very careful restrictions on what can be added there, as well as the API documentation for their use, and maybe even a short list of some features or files that are calling them.

Basically, the goal is to be able to create reports by just calling functions, and writing no new methods.

**Comment by Trodrige** [comment]
Uhmmm, I've figured out functions for providers and facilities as guided by @teryhill . I also realized that some functions already existed which displayed providers and facilities. The existing functions will be replaced with new ones which are report specific.
However, the only common date function I found is the datetimepicker JQuery code at the bottom of each report. Can anyone see other common date functions??? I'm having some issues pulling out this function to report functions file. Can you look at this @aethelwulffe , @teryhill

**Comment by teryhill** [comment]
This is for insurance companies from the collections report.

![image](https://user-images.githubusercontent.com/4655366/36881936-ee38ed40-1d9e-11e8-9719-8d7164c07da0.png)

The dates appear to be pretty compact with the date picker changes.

**Comment by Trodrige** [comment]
@teryhill did you notice this at the end of reports?
![end_of_all_reports](https://user-images.githubusercontent.com/16147419/36920899-8ae256c2-1e62-11e8-91a5-d0d25c6616d4.png)

**Comment by Trodrige** [comment]
Hello @aethelwulffe what is the use of amc_tracking in the reports? I don't how to observe the changes I make in the file.

**Comment by aethelwulffe** [comment]
AMC (Automatic Measures Calculation) is an old QA reporting system.  This, and other legacy pieces need to go away.  It needs to be part of an extensive and concentrated effort though
The new stuff is found in /module/MIPS.  Unfortunately, this module was broken by the need for the original function of the patient search/add tool to be reverted.  Newer versions of that exist, but have not been pushed to the main branch yet.  When this is done, these legacy pieces will be addressed.

**Comment by Trodrige** [comment]
@aethelwulffe @teryhill can these files be out of the reports folder for good?

1. amc_tracking
2. audit_log_tamper_report
3. cdr_log
4. cqm
5. customer_report_range
6. edi_270, edi_271, edi_276
7. inventory_activity
8. ippf_cyp_report
9. ippf_daily
10. pre_billing_issues
11. prescriptions_report
12. receipts_by_method

Made changes in these files, but couldn't find where to observe them.
